### PR TITLE
feat(ui): Make URLs clickable in workflow node info

### PR DESCRIPTION
**Pull request by Adrien-D**
No linked issue as it's a very small quick win, not sure it needs discussions

### Motivation

In workflow node details we may have templates that output links to other systems or other workflows. So that would be great to make navigation easier by showing links instead of text.

### Modifications

I implemented `linkify-it` library to find url matches and wrap those in a link tag.

| Before | After |
|---|---|
| Screenshot 2024-08-22 at 11 41 20 | Screenshot 2024-08-22 at 11 41 01 |
| Screenshot 2024-08-22 at 11 41 24 | Screenshot 2024-08-22 at 11 41 05 |

### Verification

I tested of several workflow to make sure everything is fine

### CODE REVIEW

File: ui/src/app/workflows/components/workflow-node-info/workflow-node-info.tsx [hunk 1]
```diff
@@ -68,10 +69,40 @@ interface Props {
     onResume?: () => void;
 }
 
+const linkify = new LinkifyIt();
+
+function linkifyText(text: string) {
```
Comment (agilgur5) [inline ui/src/app/workflows/components/workflow-node-info/workflow-node-info.tsx:74]: This should probably be a util function given that it's likely to be used in other places.

That also raises the question: is this the only place we'd add this? Does this open too many floodgates?

File: ui/package.json [hunk 2]
```diff
@@ -25,6 +25,7 @@
         "dagre": "^0.8.5",
         "history": "^4.10.1",
         "js-yaml": "^4.1.0",
+        "linkify-it": "^5.0.0",
```
Comment (agilgur5) [inline ui/package.json:28]: We don't seem to use all the logic from this library: https://github.com/markdown-it/linkify-it. It also doesn't seem tree-shakeable given that it relies on prototype methods

I might suggest something smaller, in the spirit of #12059

**Review by agilgur5 — COMMENTED** [review]
Oo I like the table! A few considerations below

**Comment by tooptoop4** [comment]
do u think same approach would work on https://github.com/argoproj/argo-workflows/issues/13148 ?

**Comment by agilgur5** [comment]
No, since that uses a terminal with `xterm`. If we simplified it as I [mentioned there](https://github.com/argoproj/argo-workflows/issues/13148#issuecomment-2153129909), then it could work depending on the approach

File: ui/src/app/workflows/components/workflow-node-info/workflow-node-info.tsx (hunk 1 above)
Comment (Adrien-D) [reply ui/src/app/workflows/components/workflow-node-info/workflow-node-info.tsx:74]: No you're right we could definitely use this in other places, can't see where atm but I'll make it easily reusable

File: ui/package.json (hunk 2 above)
Comment (Adrien-D) [reply ui/package.json:28]: You're right about prototype but seems to me that the impact of this library is quite negligible compared to others we could use

---- 

### Linkify-it (current) 
Linkify-it

### Linkifyjs (other I could find)
Linkifyjs

---- 
Also a note about lazy and the project.. (I wrote it here first but let's discuss it [on the issue here](https://github.com/argoproj/argo-workflows/issues/12059#issuecomment-2310473252))

Comment (Adrien-D) [reply ui/package.json:28]: I agree that the library is only partly tree shakable, but it looks quite small at the end. Should I keep looking for another lib ?

**Review by isubasinghe — APPROVED** [review]
LGTM

**Pull request merged**